File: suites/osprey-compute-kernels/reference/cases/gcdsum/gcdsum.ml
(* Sum of gcd(i, K) for i in 1..N — Euclidean-algorithm recursion (modulo heavy). *)
let rec gcd a b = if b = 0 then a else gcd b (a mod b)

let () =
  let acc = ref 0 in
  for i = 1 to 1999999 do
    acc := !acc + gcd i 1234567
  done;
  Printf.printf "%d\n" !acc

File: suites/osprey-compute-kernels/reference/cases/gcdsum/gcdsum.rs
// Sum of gcd(i, K) for i in 1..N — Euclidean-algorithm recursion (modulo heavy).
fn gcd(a: i64, b: i64) -> i64 {
    match b {
        0 => a,
        _ => gcd(b, a % b),
    }
}

fn main() {
    let mut acc: i64 = 0;
    let mut i: i64 = 1;
    while i < 2000000 {
        acc += gcd(i, 1234567);
        i += 1;
    }
    println!("{}", acc);
}
